src/actions/publish.js

```diff
@@ -98,9 +97,19 @@ export function attemptToPublish(): ThunkAction<Promise<boolean>> {
       // This way we can check inside this async function if we need to bail out early.
       const uploadGeneration = getUploadGeneration(prePublishedState);
 
-      // Create an abort function before the first async call.
+      // Create an abort function before the first async call, but we won't
+      // start the upload until much later.
       const { abortFunction, startUpload } = uploadBinaryProfileData();
-      dispatch(uploadCompressionStarted(abortFunction));
+      const augmentedAbortfunction = () => {
+        // We dispatch the action right away, so that the UI is updated.
+        // Otherwise if the user pressed "Cancel" during a long process, like
+        // the compression, we wouldn't get a feedback until the end.
+        // Later on the promise from `startUpload` will get rejected too, and we
+        // handle this in the `catch` block.
+        dispatch({ type: 'UPLOAD_ABORTED' });
+        abortFunction();
+      };
+      dispatch(uploadCompressionStarted(augmentedAbortfunction));
```

Comment:
Could we just pass in the arrow function directly without giving it a name? I found the name a bit surprising.

Reply:
I renamed the property returned by uploadBinaryProfileData to `abortUpload` (this makes sense because the other function is `startUpload`. Then I can call this local variable simply `abortFunction` which is the name of the state.
I found that it's better to name the function because then we understand more easily what is this thing we pass to this action.